### DISCUSSION TITLE
Action naming convention

### DISCUSSION THREAD

**Discussion by davdroman** [pointfreeco/swift-composable-architecture#1241] [Q&A] [answered]
A small question I have from seeing https://github.com/pointfreeco/swift-composable-architecture/pull/1240

What's the rationale for action names being written in the past participle tense instead of the present tense?

```swift
case alertDismissed
// vs
case dismissAlert
```

**Comment by tgrapperon** [comment, marked as answer]
I think it serves two different purposes. The first one is "informative", it says that an event has occurred and an action has been emitted accordingly, and you should perform relevant tasks in your reducer when you receive it, like cleaning up some state.
The second one is "imperative". The action bears an intent to do something. When you receive it in your reducer, you should perform the tasks required to realize the action.

In this specific case, this is a little tricky. The `.dismiss` action is emitted by the framework when the alert is dismissed by the system. This is not a request to dismiss the alert, but a signal emitted by the system that the user has finished interacting with the alert without using a button. In in sense, `.alertDismissed` is indeed probably more correct in this situation, as I think (but I would need to check) that the alert will be dismissed even if the state is not nil'ed.

I personally think that both names are correct and can offer some contextual nuance in complex domains. I'm personally mixing both styles, as I feel that semantic correctness is preferable to stylistic consistency, and I feel very smug while writing this. 

It is even possible to imagine domains with a `.dismissView` and a `.viewDismissed` action.
